### ISSUE TITLE
ROUNDDOWN calculation wrong

### ISSUE THREAD

**Issue by longcharmroeun** [EPPlusSoftware/EPPlus#1087]
### EPPlus usage

Personal use

### Environment

window

### Epplus version

6.2.10

### Spreadsheet application

Excel

### Description

```
using (var package = new ExcelPackage())
            {
                var sheet = package.Workbook.Worksheets.Add("test");
                sheet.Cells["A1"].Value = 17.38;
                sheet.Cells["A2"].Formula = "ROUNDDOWN(A1,2)";
                sheet.Calculate(opt => opt.PrecisionAndRoundingStrategy = PrecisionAndRoundingStrategy.Excel);
                var resulf = sheet.Cells["A2"].ToText();
                //result should be 17.38 but after calculate. result display 17.37
            }
```

**Comment by OssianEPPlus** [comment]
Yep. That looks like a floating point bug. Will be looking into it

**Comment by OssianEPPlus** [comment]
We've resolved this bug a while back. The fix will be released in a new version either this or next week. If you wish you could test the fix now via our [Develop NuGet Feed](https://github.com/EPPlusSoftware/EPPlus/wiki/Using-our-develop-Nuget-feed)

Thanks for bringing this to our attention!

**Comment by longcharmroeun** [comment]
```
using (var package = new ExcelPackage())
            {
                var sheet = package.Workbook.Worksheets.Add("test");
                sheet.Cells["A1"].Value = 17.38;
                sheet.Cells["A2"].Formula = "ROUNDDOWN(A1,2)";
                sheet.Calculate(opt => opt.PrecisionAndRoundingStrategy = PrecisionAndRoundingStrategy.Excel);
                var resulf = sheet.Cells["A2"].ToText();
                //result should be 17.38 but after calculate. result display 17.37
            }
```
This example bug is already fixed but i got an other similar bug as below

```
using (var package = new ExcelPackage())
            {
                var sheet = package.Workbook.Worksheets.Add("test");
                sheet.Cells["A1"].Value = 2347440000d;
                sheet.Cells["A2"].Formula = "ROUNDDOWN(A1,-5)";
                sheet.Calculate(opt => opt.PrecisionAndRoundingStrategy = PrecisionAndRoundingStrategy.Excel);
                var resulf = sheet.Cells["A2"].ToText();
                //result should be 2347400000 but after calculate. result display -2147400000
            }
```

**Comment by OssianEPPlus** [comment]
Ah, Good catch! Thank you. Looks like we were casting parts of the number to int for internal operations which didn't work for large numbers. A fix is on the way.

**Comment by OssianEPPlus** [comment]
Resolved as of epplus 7.0 and 6.2.11

**Comment by OssianEPPlus** [comment]
-
